### ISSUE TITLE
Metabolites with whitespace in ID are accepted but cause errors after copying the Model

### ISSUE THREAD

**Issue by eladnoor** [opencobra/cobrapy#690]
#### Problem description

I tried to use the flux_analysis.sample function on my model, which yielded an error regarding the array of variables (which was empty). I traced the bug back to the copy() function (which is used by the sampler) where the solver is copied using the copy.deepcopy() function. For GLPK, this copy fails when there are metabolites with a space in the middle of their ID.
This bug does not occur if:
* the new metabolite ID has a single sequence of alphanumeric characters (even if there are trailing spaces before or after it)
* the solver is not GLPK (e.g. CPLEX)

OS: Ubuntu 17.10, Python: 3.6, cobrapy: 0.11.3

#### Code Sample

```python
from cobra import Metabolite
from cobra.test import create_test_model
from copy import deepcopy

model = create_test_model()
model.add_metabolites([Metabolite('A A')])
s = deepcopy(model.solver)
print(len(s.variables))
```

#### Actual Output
0

#### Expected Output
5092

#### Output of `cobra.show_versions()`

<details>
System Information
==================
OS                         Linux
OS-release     4.13.0-38-generic
Python                     3.6.4

Package Versions
================
pip                        9.0.3
setuptools                38.6.0
cobra                     0.11.3
future                    0.16.0
swiglpk                    1.4.4
optlang                    1.4.0
ruamel.yaml              0.14.12
pandas                    0.22.0
numpy                     1.14.2
tabulate                   0.8.1
python-libsbml            5.16.0
lxml                       4.1.1
scipy                      1.0.0
matplotlib                 2.2.0
palettable                 3.1.0
</details>

**Comment by cdiener** [comment]
I suspect that is because optlang serializes the C solver object by writing and re-reading it from a .lp file. You can probably not have spaces there since it would look something like

```
var_1 + A A <= 1000
``` 

which might be a syntax error. Will report it to optlang.

**Comment by Midnighter** [comment]
I wanted to look up what we do when instantiating optlang variables since optlang actually does not allow for spaces in variable names.

**Comment by eladnoor** [comment]
As a workaround, you could simply not allow spaces in metabolite IDs, same as for reaction IDs.

**Comment by cdiener** [comment]
@Midnighter I think it affects the constraint names here and those can have spaces as it seems:

```Python
In [1]: import optlang.glpk_interface as prob

In [2]: mod = prob.Model()

In [3]: const = prob.Constraint(1, name="with spaces ")

In [4]: mod.add(const)

In [5]: mod.optimize()
Out[5]: 'optimal'
```

**Comment by Midnighter** [comment]
Doh, of course. Thank you for pointing that out. Let's handle it in optlang as you suggested.

**Comment by cdiener** [comment]
Fixed with optlang 1.4.1. Now throws an error.
